yb-voyager/cmd/exportData.go

```diff
@@ -90,35 +93,27 @@ func exportDataOffline() bool {
 	// defer cancel()
 
 	var tableList []string
+	var finalTableList []string
+	excludeTableList := extractTableListFromString(source.ExcludeTableList) //function handles null case
 	if source.TableList != "" {
-		userTableList := utils.CsvStringToSlice(source.TableList)
+		finalTableList = extractTableListFromString(source.TableList)
+	} else {
 
-		if source.DBType == POSTGRESQL {
-			// in postgres format should be schema.table, public is default and other parts of code assume schema.table format
-			for _, table := range userTableList {
-				parts := strings.Split(table, ".")
-				if len(parts) == 1 {
-					tableList = append(tableList, "public."+table)
-				} else if len(parts) == 2 {
-					tableList = append(tableList, table)
-				} else {
-					utils.ErrExit("invalid table name %q in the --table-list flag.", table)
+		tableList = source.DB().GetAllTableNames()
+		if len(excludeTableList) > 0 {
```

Comment:
This check is not required. For empty slice or nil, the for loop simply doesn't execute the loop's body.

Reply:
The slices are different right? The for loop uses the tableList which we have to export, while the excludeTableList is the table lists which we have to remove
Still, if we remove the "if" check the end behaviour is the same, so will do it